### PR TITLE
feat/CXSPA-6636: OPPS Coupon Codes & Login required

### PR DESCRIPTION
**Pull request by anjana-bl**
Follow up of https://github.com/SAP/spartacus/pull/18727

### CODE REVIEW

File: integration-libs/opps/root/coupon-codes/opps-coupon-codes.service.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  getCouponCodes(): string | undefined | null {
+    if (this.localStorageKey) {
+      return this.winRef.localStorage?.getItem(this.localStorageKey);
+    }
+  }
+  clearCouponCodes(): void {
+    if (this.localStorageKey) {
+      this.winRef.localStorage?.removeItem(this.localStorageKey);
+    }
+  }
```
Comment (Platonn) [inline integration-libs/opps/root/coupon-codes/opps-coupon-codes.service.ts]: I have problem to find usages of this method

File: integration-libs/opps/root/coupon-codes/opps-coupon-codes.module.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+import { NgModule, inject } from '@angular/core';
+import { LOCATION_INITIALIZED_MULTI } from '@spartacus/core';
+import { oppsCouponCodesInterceptors } from './http-interceptors';
+import { OppsCouponCodesService } from './opps-coupon-codes.service';
+
+export function saveCouponCodesFactory(): () => void {
+  return () => {
+    const service = inject(OppsCouponCodesService);
```
Comment (Platonn) [inline integration-libs/opps/root/coupon-codes/opps-coupon-codes.module.ts]: Hmmm. In APP_INITIALIZERS you _have to_ inject stuff outside of the returned callback. But this is `LOCATION_INITIALIZED_MULTI` hook and perhaps here it's not forced.

But for consistency and readability, I recommend to do the same here -> let's swap above 2 lines.
Let's first let's inject all depenencies. Then let's return the callback.

File: projects/schematics/package.json [hunk 3]
```diff
@@ -13,8 +13,8 @@
   "scripts": {
     "build": "npm run clean && npm --prefix ../../ run generate:deps && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
     "clean": "../../node_modules/.bin/rimraf --glob \"index.js*\" \"index.d.ts\" \"src/**/*.js\" \"src/**/*.js.map\" \"src/**/*.d.ts\" && npm run copy-feature-toggles",
-    "test": "npm run clean && ../../node_modules/.bin/jest --config ./jest.schematics.config.js",
-    "copy-feature-toggles": "node copy-feature-toggles"
```
Comment (Platonn) [inline projects/schematics/package.json:17]: Is this change _really_ related to this PR?

File: projects/schematics/src/shared/lib-configs/integration-libs/opps-schematics-config.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  featureModule: {
+    name: OPPS_ROOT_MODULE,
+    importPath: SPARTACUS_OPPS,
+  },
+  rootModule: {
+    importPath: SPARTACUS_OPPS_ROOT,
+    name: OPPS_ROOT_MODULE,
+    content: `${OPPS_ROOT_MODULE}`,
+  },
+  /* Through Spartacus OPPS code doesnot have a dependency on Personalization,
```
Comment (Platonn) [inline projects/schematics/src/shared/lib-configs/integration-libs/opps-schematics-config.ts]:
```suggestion
  /* Through Spartacus OPPS code does not have a dependency on Personalization,
```

File: integration-libs/opps/root/coupon-codes/opps-coupon-codes.service.ts (hunk 1 above)
Comment (anjana-bl) [reply integration-libs/opps/root/coupon-codes/opps-coupon-codes.service.ts]: Hi Kris, it is decided by PO to leave the coupon codes in browser without clearing. Hence this method is not used. i will remove this method.

File: integration-libs/opps/root/coupon-codes/opps-coupon-codes.module.ts (hunk 2 above)
Comment (anjana-bl) [reply integration-libs/opps/root/coupon-codes/opps-coupon-codes.module.ts]: yes done

File: projects/schematics/package.json (hunk 3 above)
Comment (anjana-bl) [reply projects/schematics/package.json:17]: not needed, committed by mistake

File: integration-libs/opps/root/login-required/index.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export * from './opps-login-required.module';
```
Comment (Platonn) [inline integration-libs/opps/root/login-required/index.ts:7]: For customers for extensibility, I believe we want to also export the `guard` file

File: integration-libs/opps/root/coupon-codes/http-interceptors/index.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Provider } from '@angular/core';
+import { HTTP_INTERCEPTORS } from '@angular/common/http';
+import { OccOppsCouponCodesInterceptor } from './occ-opps-coupon-codes.interceptor';
+
```
Comment (Platonn) [inline integration-libs/opps/root/coupon-codes/http-interceptors/index.ts:10]: For customers for extensibility let's re-rexport here also the `interceptor` class
`export * from './occ-opps-coupon-codes.interceptor';`

**Review by Platonn — DISMISSED** [review]

File: integration-libs/opps/root/login-required/index.ts (hunk 5 above)
Comment (anjana-bl) [reply integration-libs/opps/root/login-required/index.ts:7]: done

File: integration-libs/opps/root/coupon-codes/http-interceptors/index.ts (hunk 6 above)
Comment (anjana-bl) [reply integration-libs/opps/root/coupon-codes/http-interceptors/index.ts:10]: done

**Review by Platonn — APPROVED** [review]
